docs/operators/regexparser.md

```diff
@@ -0,0 +1,129 @@
+Regex Parser Operator
+=====================
+
+## Operator Objective
+This operator is designed to parse records based on a regex pattern and construct a concrete java class also known as ["POJO"](https://en.wikipedia.org/wiki/Plain_Old_Java_Object) out of it. User need to provide the regex pattern and schema definition to describe the data pattern. Based on regex pattern, the operator will split the data and then schema definition will be used to map the incoming record to POJO. User can also provide date format if any, in the schema. The supported constraints are listed in [constraints table](#constraints). Valid records will be emitted as POJO while invalid ones are emitted on error port with error message.
+
+**Note**: field names of POJO must match field names in schema and in the same order as it appears in the incoming data.
+
+## Overview
+The operator is **idempotent**, **fault-tolerant** and **partitionable**.
+
+## Operator Information
+1. Operator location:**_malhar-contrib_**
+2. Available since:**_3.7.0_**
+3. Operator state:**_Evolving_**
+4. Java Package:[com.datatorrent.contrib.parser.RegexParser](https://github.com/apache/apex-malhar/blob/master/contrib/src/main/java/com/datatorrent/contrib/parser/RegexParser.java)
+
+
+## <a name="props"></a>Properties of Regex Parser
+User need to set the schema string which describes data schema as well as specify the format on date fields if any.
+e.g.
+
+```xml
+{
+   "fields": [
+      {
+        "name": "date",
+        "type": "Date",
+        "constraints": {
+          "format": "yyyy:MM:dd:hh:mm:ss"
+         }
+      },
+      {
+        "name": "id",
+        "type": "Integer"
+      },
+      {
+        "name": "message",
+        "type": "String"
+      }
+   ]
+}
+```
+
+User need to set the splitRegexPattern which is the regex expression that describes the pattern of the incoming data.
```

Comment:
need => needs
splitRegexPattern which is the regex => `splitRegexPattern` property whose value is the regular